cas-server-core-tickets/src/main/java/org/jasig/cas/ticket/AbstractTicket.java

```diff
@@ -151,12 +153,29 @@ protected boolean isExpiredInternal() {
     }
 
     @Override
-    public final int hashCode() {
-        return new HashCodeBuilder(13, 133).append(this.getId()).toHashCode();
+    public final String toString() {
+        return this.getId();
     }
 
     @Override
-    public final String toString() {
-        return this.getId();
+    public boolean equals(final Object o) {
```

Comment:
Unfortunately, I don't recall the exact case, but moving this upwards will cause issues later on. I know this because at some point not too long ago I attempted to do the same exact thing. It may work now that we have made a few API changes, but let's play it safe and revisit later.